kafka-rest/src/main/java/io/confluent/kafkarest/controllers/ConsumerGroupLagManagerImpl.java

```diff
@@ -0,0 +1,118 @@
+    this.consumerGroupManager = requireNonNull(consumerGroupManager);
+  }
+
+  @Override
+  public CompletableFuture<Optional<ConsumerGroupLag>> getConsumerGroupLag(
+      String clusterId,
+      String consumerGroupId
+  ) {
+    return consumerGroupManager.getConsumerGroup(clusterId, consumerGroupId)
+        .thenApply(
+            consumerGroup ->
+                checkEntityExists(consumerGroup, "Consumer Group %s could not be found.", consumerGroupId))
+        .thenCompose(
+            consumerGroup ->
+                getCurrentOffsets(consumerGroupId)
+                    .thenApply(
+                        fetchedCurrentOffsets ->
+                            checkOffsetsExist(fetchedCurrentOffsets, "Consumer group offsets could not be found."))
+                    .thenCompose(
+                        fetchedCurrentOffsets ->
+                            getLatestOffsets(fetchedCurrentOffsets)
+                                .thenApply(
+                                    latestOffsets ->
+                                       createConsumerGroupLag(
+                                           clusterId,
+                                           consumerGroup,
+                                           fetchedCurrentOffsets,
+                                           latestOffsets))));
+  }
+
+  Optional<ConsumerGroupLag> createConsumerGroupLag(
+      String clusterId,
+      ConsumerGroup consumerGroup,
+      Map<TopicPartition, OffsetAndMetadata> fetchedCurrentOffsets,
+      Map<TopicPartition, ListOffsetsResultInfo> latestOffsets) {
+    Map<TopicPartition, MemberId> tpMemberIds =
+        getMemberIds(consumerGroup);
+    ConsumerGroupLag.Builder consumerGroupLag =
+        ConsumerGroupLag.builder()
+            .setClusterId(clusterId)
+            .setConsumerGroupId(consumerGroup.getConsumerGroupId());
+    fetchedCurrentOffsets.keySet().stream().forEach(
+        topicPartition -> {
+          MemberId memberId = tpMemberIds.getOrDefault(
+              topicPartition, MemberId.builder()
+                  .setConsumerId("")
+                  .setClientId("")
+                  .setInstanceId("")
+                  .build());
+          long currentOffset =
+              getCurrentOffset(fetchedCurrentOffsets, topicPartition);
+          long latestOffset =
+              getOffset(latestOffsets, topicPartition);
+          if (currentOffset < 0 || latestOffset < 0) {
+            log.debug("invalid offsets for topic={} partition={} consumerId={} current={} latest={}",
+                topicPartition.topic(),
+                topicPartition.partition(),
+                memberId.getConsumerId(),
+                currentOffset,
+                latestOffset);
+          } else {
+            consumerGroupLag.addOffset(
+                topicPartition.topic(),
+                memberId.getConsumerId(),
+                memberId.getClientId(),
+                memberId.getInstanceId().orElseGet(null),
```

Comment:
This will cause a `NullPointerException`. `orElseGet` receives a `Supplier<T>`, to which the Optional will call `get()` to obtain the value to be used. Here you are passing a null `Supplier`, so `supplier.get()` will throw `NullPointerException`. What you probably want is `orElse(null)`.